# Objective

The main objective of this project is to address critical challenges in shipment and inventory management, identify inefficiencies in the supply chain, and develop informative dashboards. By doing so, I aim to provide valuable insights to business stakeholders, enabling them to recognize potential issues and suggest necessary improvements to the overall business operations.

To achieve this, I have performed data preprocessing in Google Colab before proceeding with the analysis in Power BI. Through data cleaning, validation, and anomaly detection, I have prepared the data for further exploration and visualization. With these efforts, I hope to contribute to more efficient and effective supply chain management, leading to better decision-making and enhanced business performance.

"*Garbage in, garbage out*" is a simple phrase that highlights the importance of using good quality data for accurate results. It means that if we use bad or flawed data for analysis, we will get unreliable or incorrect outcomes. In other words, the conclusions we draw from data analysis can only be as good as the data we put into it. Therefore, it's essential to ensure that our data is clean, accurate, and reliable before conducting any analysis to make sure we get meaningful and trustworthy insights.

# Data Preparation

In [ ]:
# import the libraries
import pandas as pd
import numpy as np

In [ ]:
# import datasets
df_orders = pd.read_csv('orders_and_shipments.csv')
df_inventory = pd.read_csv('inventory.csv')
df_fulfillment = pd.read_csv('fulfillment.csv')

In [ ]:
df_orders.sample(5)

Order ID    Order Item ID    Order YearMonth    Order Year   \
20237      49988           124968             201703          2017   
18124      39623            98898             201608          2016   
2585       65353           163349             201708          2017   
18235      15378            38460             201601          2016   
25717      11611            29031             201506          2015   

        Order Month    Order Day  Order Time  Order Quantity  \
20237              3           29      14:58               1   
18124              8            1      09:21               1   
2585               8           11      23:42               5   
18235              1            9      04:33               1   
25717              6           19      11:30               1   

      Product Department      Product Category  ... Customer Country  \
20237            Apparel        Men's Footwear  ...          Nigeria   
18124           Fan Shop      Camping & Hiking  ...              USA   
2585             Apparel                Cleats  ...   United Kingdom   
18235           Fan Shop      Camping & Hiking  ...          Germany   
25717           Fan Shop  Indoor/Outdoor Games  ...           France   

       Warehouse Country Shipment Year Shipment Month Shipment Day  \
20237                USA          2017              4            4   
18124                USA          2016              8            5   
2585                 USA          2017              8           15   
18235                USA          2017              9           21   
25717        Puerto Rico          2015              6           21   

        Shipment Mode   Shipment Days - Scheduled    Gross Sales   \
20237  Standard Class                            4            130   
18124    Second Class                            2            300   
2585     Second Class                            2            300   
18235  Standard Class                            4            300   
25717  Standard Class                            4             50   

        Discount %   Profit   
20237           0.2       65  
18124          0.07      150  
2585           0.02      147  
18235          0.01      150  
25717          0.13      125  

[5 rows x 24 columns]

In [ ]:
df_inventory.sample(5)

Product Name   Year Month   \
1582  Bridgestone e6 Straight Distance NFL Tennesse        201611   
4093          The North Face Women's Recon Backpack        201502   
2074                                           Toys        201607   
137         Columbia Men's PFG Anchor Tough T-Shirt        201711   
2063                               Porcelain crafts        201607   

       Warehouse Inventory   Inventory Cost Per Unit  
1582                      1                  1.96084  
4093                      0                  1.93772  
2074                      0                  1.43895  
137                       0                  1.37329  
2063                      0                  1.26192

In [ ]:
df_fulfillment.sample(5)

Product Name  \
34                      TYR Boys' Team Digi Jammer   
86   Merrell Men's All Out Flash Trail Running Sho   
101                        GolfBuddy VT3 GPS Watch   
30       Team Golf Pittsburgh Steelers Putter Grip   
33                   Nike Dri-FIT Crew Sock 6 Pack   

      Warehouse Order Fulfillment (days)   
34                                    8.0  
86                                    6.8  
101                                   6.7  
30                                    9.1  
33                                    6.8

# Data Cleaning

## Handling Missing Value

In [ ]:
df_orders.isna().sum()

Order ID                       0
 Order Item ID                 0
 Order YearMonth               0
 Order Year                    0
 Order Month                   0
 Order Day                     0
Order Time                     0
Order Quantity                 0
Product Department             0
Product Category               0
Product Name                   0
 Customer ID                   0
Customer Market                0
Customer Region                0
Customer Country               0
Warehouse Country              0
Shipment Year                  0
Shipment Month                 0
Shipment Day                   0
Shipment Mode                  0
 Shipment Days - Scheduled     0
 Gross Sales                   0
 Discount %                    0
 Profit                        0
dtype: int64

In [ ]:
df_inventory.isna().sum()

Product Name               0
 Year Month                0
 Warehouse Inventory       0
Inventory Cost Per Unit    0
dtype: int64

In [ ]:
df_fulfillment.isna().sum()

Product Name                            0
 Warehouse Order Fulfillment (days)     0
dtype: int64

However, upon inspection, it was found that all three datasets do not contain any missing values. This is a good sign, as it ensures the data's completeness and saves us from the need to handle missing data imputation.

## Duplicated Data

In [ ]:
df_orders.duplicated().sum()

0

In [ ]:
df_inventory.duplicated().sum()

0

In [ ]:
df_fulfillment.duplicated().sum()

0

Moreover, the absence of any duplicate values in all three datasets is also promising. This indicates that the data is fairly clean and ready for analysis.

# Data Wrangling

In [ ]:
# Replace 'LATAM' with 'Latin America' in the 'Customer Market' column
df_orders['Customer Market'] = df_orders['Customer Market'].replace('LATAM', 'Latin America')

Now, the data in the "`Customer Market`" column will be easier to understand, as it will represent the geographic grouping of customer countries as "`Latin America`" instead of the abbreviation "`LATAM`".

## Leading and Trailing Space

In [ ]:
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30871 entries, 0 to 30870
Data columns (total 24 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Order ID                     30871 non-null  int64 
 1    Order Item ID               30871 non-null  int64 
 2    Order YearMonth             30871 non-null  int64 
 3    Order Year                  30871 non-null  int64 
 4    Order Month                 30871 non-null  int64 
 5    Order Day                   30871 non-null  int64 
 6   Order Time                   30871 non-null  object
 7   Order Quantity               30871 non-null  int64 
 8   Product Department           30871 non-null  object
 9   Product Category             30871 non-null  object
 10  Product Name                 30871 non-null  object
 11   Customer ID                 30871 non-null  int64 
 12  Customer Market              30871 non-null  object
 13  Customer Region              30

In [ ]:
df_inventory.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4200 entries, 0 to 4199
Data columns (total 4 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Product Name             4200 non-null   object 
 1    Year Month              4200 non-null   int64  
 2    Warehouse Inventory     4200 non-null   int64  
 3   Inventory Cost Per Unit  4200 non-null   float64
dtypes: float64(1), int64(2), object(1)
memory usage: 131.4+ KB


In [ ]:
df_fulfillment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118 entries, 0 to 117
Data columns (total 2 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Product Name                          118 non-null    object 
 1    Warehouse Order Fulfillment (days)   118 non-null    float64
dtypes: float64(1), object(1)
memory usage: 2.0+ KB


Upon inspecting the data using `.info()`, it has come to our attention that there are anomalies present in the datasets. Specifically, we have discovered that some columns contain leading or trailing spaces, which are not intended to be part of the column names.

In [ ]:
dataframes = [df_inventory, df_orders, df_fulfillment]

for df in dataframes:
  df.columns = df.columns.str.strip()

In [ ]:
df_orders.columns

Index(['Order ID', 'Order Item ID', 'Order YearMonth', 'Order Year',
       'Order Month', 'Order Day', 'Order Time', 'Order Quantity',
       'Product Department', 'Product Category', 'Product Name', 'Customer ID',
       'Customer Market', 'Customer Region', 'Customer Country',
       'Warehouse Country', 'Shipment Year', 'Shipment Month', 'Shipment Day',
       'Shipment Mode', 'Shipment Days - Scheduled', 'Gross Sales',
       'Discount %', 'Profit'],
      dtype='object')

In [ ]:
df_inventory.columns

Index(['Product Name', 'Year Month', 'Warehouse Inventory',
       'Inventory Cost Per Unit'],
      dtype='object')

In [ ]:
df_fulfillment.columns

Index(['Product Name', 'Warehouse Order Fulfillment (days)'], dtype='object')

In [ ]:
# Convert the '-' values to 0 in the 'Discount %' column and then change the data type from object to float
df_orders['Discount %'] = df_orders['Discount %'].replace('  -  ', 0).astype(float)

## Order Datetime & Shipment Datetime

Due to the data being separated into multiple columns for year, month, day, and time, resulting in a large number of columns, I have decided to combine these columns into a single column that includes all the necessary date and time information. This will help simplify the data and make it more manageable for further analysis in Power BI.

In [ ]:
# Make new columns: Order Datetime and Shipment Datetime
df_orders['Order Datetime'] = pd.to_datetime(df_orders['Order Year'].astype(str) + '-' + df_orders['Order Month'].astype(str) + '-' + df_orders['Order Day'].astype(str) + ' ' + df_orders['Order Time'])
df_orders['Shipment Datetime'] = pd.to_datetime(df_orders['Shipment Year'].astype(str) + '-' + df_orders['Shipment Month'].astype(str) + '-' + df_orders['Shipment Day'].astype(str))

# Displaying the result
df_orders[['Order Datetime', 'Shipment Datetime']].head()

Order Datetime Shipment Datetime
0 2015-02-21 14:07:00        2015-02-27
1 2015-03-02 07:37:00        2015-03-06
2 2015-04-18 22:47:00        2015-04-20
3 2015-06-10 22:32:00        2015-06-12
4 2015-06-10 22:32:00        2015-06-12

In [ ]:
# Drop unnecessary columns
df_orders.drop(columns=['Order Year', 'Order Month', 'Order Day', 'Order Time',
                        'Shipment Year', 'Shipment Month', 'Shipment Day'], inplace=True)

# Data Manipulation

## Order Processing Time

The "`Order Processing Time`" is obtained by subtracting the "`Order Datetime`" from the "`Shipment Datetime`". It measures the time taken for the order to move through various stages, including processing, packing, and delivery preparation, until it is ready for shipment. This metric helps us understand how quickly we can fulfill customer orders and deliver products to their destination.

In [ ]:
df_orders['Order Processing Time'] = (df_orders['Shipment Datetime'] - df_orders['Order Datetime']).dt.days
df_orders['Order Processing Time'] = df_orders['Order Processing Time'].apply(lambda x: 0 if x == -1 else x)

df_orders.sample(5)

Order ID  Order Item ID  Order YearMonth  Order Quantity  \
18938     11371          28437           201506               1   
8203      67901         169773           201709               1   
19494     41553         103727           201701               1   
25302     35925          89721           201606               1   
17434     55263         138211           201703               1   

      Product Department      Product Category  \
18938           Fan Shop  Indoor/Outdoor Games   
8203            Fan Shop          Water Sports   
19494           Fan Shop          Water Sports   
25302           Fan Shop          Water Sports   
17434           Fan Shop               Fishing   

                                    Product Name  Customer ID Customer Market  \
18938           O'Brien Men's Neoprene Life Vest         5076          Europe   
8203                 Pelican Maverick 100X Kayak         5646          Europe   
19494                Pelican Sunstream 100 Kayak         1793    Pacific Asia   
25302                Pelican Sunstream 100 Kayak        11318   North America   
17434  Field & Stream Sportsman 16 Gun Fire Safe        12087   Latin America   

       Customer Region Customer Country Warehouse Country   Shipment Mode  \
18938   Western Europe      Netherlands               USA    Second Class   
8203   Southern Europe            Spain               USA  Standard Class   
19494        West Asia     Saudi Arabia               USA    Second Class   
25302      East of USA              USA       Puerto Rico        Same Day   
17434  Central America           Mexico               USA  Standard Class   

       Shipment Days - Scheduled  Gross Sales  Discount %  Profit  \
18938                          2           50        0.13     125   
8203                           4          350        0.01     172   
19494                          2          200        0.13     100   
25302                          3          200        0.10     100   
17434                          4          400        0.13     200   

           Order Datetime Shipment Datetime  Order Processing Time  
18938 2015-06-15 23:25:00        2015-06-17                      1  
8203  2017-09-18 04:23:00        2017-09-20                      1  
19494 2017-01-06 03:31:00        2017-01-12                      5  
25302 2016-06-08 09:46:00        2016-06-08                      0  
17434 2017-03-17 16:44:00        2017-03-21                      3

The presence of timestamps in the order of dates provides more accurate and detailed information about the time intervals between events. However, when calculating "`Order Processing Time`" using a timestamp, it can result in a negative value of 1 on same-day shipping as the order, because the shipping date does not have a timestamp, which is why the default timestamp of 00:00:00 is used.

To address this issue, we implemented a `lambda` function to set a negative value of 1 to 0, ensuring the calculated duration is not negative and meaningful for analysis. This approach allows us to gain valuable insights into the efficiency of the shipping process and delivery time, taking into account the time-sensitive nature of the data.

## Stock Coverage Ratio

The formula to calculate the Stock Coverage Ratio is `Warehouse Inventory` divided by Average Demand.

In simple terms, the Stock Coverage Ratio indicates how many days or units of inventory are available in the warehouse to fulfill customer orders, considering the average demand for products during a particular time frame.

In [ ]:
# First, create the average demand
average_demand = df_orders.groupby('Product Name')['Order Quantity'].mean().reset_index()
average_demand.rename(columns={'Order Quantity': 'Average Demand'}, inplace=True)
average_demand

Product Name  Average Demand
0                               Adult dog supplies        1.000000
1                                     Baby sweater        1.000000
2                          Bag Boy Beverage Holder        3.061224
3                           Bag Boy M330 Push Cart        3.100000
4    Bridgestone e6 Straight Distance NFL Carolina        3.022727
..                                             ...             ...
108          adidas Kids' F5 Messi FG Soccer Cleat        2.589744
109     adidas Men's F10 Messi TRX FG Soccer Cleat        3.047619
110      adidas Men's Germany Black Crest Away Tee        2.840909
111  adidas Youth Germany Black/Red Away Match Soc        3.339623
112               insta-bed Neverflat Air Mattress        1.000000

[113 rows x 2 columns]

In [ ]:
# Because the Warehouse Inventory is in df_inventory, a join table is needed
merged_inventory = df_inventory.merge(average_demand, on='Product Name', how='left')
merged_inventory.head()

Product Name  Year Month  \
0               Perfect Fitness Perfect Rip Deck      201712   
1           Nike Men's Dri-FIT Victory Golf Polo      201712   
2               O'Brien Men's Neoprene Life Vest      201712   
3              Nike Men's Free 5.0+ Running Shoe      201712   
4  Under Armour Girls' Toddler Spine Surge Runni      201712   

   Warehouse Inventory  Inventory Cost Per Unit  Average Demand  
0                    0                  0.69517        3.017707  
1                    2                  1.29291        3.005522  
2                    0                  0.56531        2.975410  
3                    1                  1.26321        3.036218  
4                    0                  1.47648        3.058602

In [ ]:
# Calculate the Stock Coverage Ratio
merged_inventory['Stock Coverage Ratio'] = merged_inventory['Warehouse Inventory'] / merged_inventory['Average Demand']
merged_inventory.head()

Product Name  Year Month  \
0               Perfect Fitness Perfect Rip Deck      201712   
1           Nike Men's Dri-FIT Victory Golf Polo      201712   
2               O'Brien Men's Neoprene Life Vest      201712   
3              Nike Men's Free 5.0+ Running Shoe      201712   
4  Under Armour Girls' Toddler Spine Surge Runni      201712   

   Warehouse Inventory  Inventory Cost Per Unit  Average Demand  \
0                    0                  0.69517        3.017707   
1                    2                  1.29291        3.005522   
2                    0                  0.56531        2.975410   
3                    1                  1.26321        3.036218   
4                    0                  1.47648        3.058602   

   Stock Coverage Ratio  
0              0.000000  
1              0.665442  
2              0.000000  
3              0.329357  
4              0.000000

## Total Inventory Cost

Total Cost of Inventory is a significant financial metric in supply chain management that calculates the overall cost associated with holding and managing inventory.

The formula for calculating Total Inventory Cost is: `Warehouse Inventory` times `Inventory Cost per Unit`.

In [ ]:
merged_inventory['Total Inventory Cost'] = merged_inventory['Warehouse Inventory'] * merged_inventory['Inventory Cost Per Unit']
merged_inventory.head()

Product Name  Year Month  \
0               Perfect Fitness Perfect Rip Deck      201712   
1           Nike Men's Dri-FIT Victory Golf Polo      201712   
2               O'Brien Men's Neoprene Life Vest      201712   
3              Nike Men's Free 5.0+ Running Shoe      201712   
4  Under Armour Girls' Toddler Spine Surge Runni      201712   

   Warehouse Inventory  Inventory Cost Per Unit  Average Demand  \
0                    0                  0.69517        3.017707   
1                    2                  1.29291        3.005522   
2                    0                  0.56531        2.975410   
3                    1                  1.26321        3.036218   
4                    0                  1.47648        3.058602   

   Stock Coverage Ratio  Total Inventory Cost  
0              0.000000               0.00000  
1              0.665442               2.58582  
2              0.000000               0.00000  
3              0.329357               1.26321  
4              0.000000               0.00000

In [ ]:
merged_inventory.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4200 entries, 0 to 4199
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Product Name             4200 non-null   object 
 1   Year Month               4200 non-null   int64  
 2   Warehouse Inventory      4200 non-null   int64  
 3   Inventory Cost Per Unit  4200 non-null   float64
 4   Average Demand           4025 non-null   float64
 5   Stock Coverage Ratio     4025 non-null   float64
 6   Total Inventory Cost     4200 non-null   float64
dtypes: float64(4), int64(2), object(1)
memory usage: 262.5+ KB


In the `merged_inventory` table, the `null` data represents products that have not been ordered at all in the `df_orders` table. These products have not been part of any transaction and, therefore, do not have any corresponding order information in the `df_orders` dataset.

## Shipment Delay

Shipment Delay is a metric that measures the time difference between the expected shipment date and the actual date that the order is delivered to the customer. This helps identify and measure the efficiency and reliability of the shipping processes.

The calculation of Shipment Delay involves comparing the `Delivery Date` (actual date the order is shipped) with the `Shipment Days - Scheduled` (the expected or planned date of shipment).

In [ ]:
df_orders['Shipment Delay'] = df_orders['Shipment Days - Scheduled'] - df_orders['Order Processing Time']
df_orders[['Order Datetime', 'Shipment Days - Scheduled', 'Shipment Datetime', 'Order Processing Time', 'Shipment Delay']].head()

Order Datetime  Shipment Days - Scheduled Shipment Datetime  \
0 2015-02-21 14:07:00                          4        2015-02-27   
1 2015-03-02 07:37:00                          4        2015-03-06   
2 2015-04-18 22:47:00                          4        2015-04-20   
3 2015-06-10 22:32:00                          4        2015-06-12   
4 2015-06-10 22:32:00                          4        2015-06-12   

   Order Processing Time  Shipment Delay  
0                      5              -1  
1                      3               1  
2                      1               3  
3                      1               3  
4                      1               3

If the product is shipped later than the expected schedule, the "`Shipment Delay`" will be represented by a negative value, indicating a delay in the delivery. On the other hand, if the product is shipped earlier than the scheduled date, the "`Shipment Delay`" will be positive, signifying an early delivery compared to the scheduled timeframe.

# Exploratory Data Analysis

In [ ]:
df_orders.columns

Index(['Order ID', 'Order Item ID', 'Order YearMonth', 'Order Quantity',
       'Product Department', 'Product Category', 'Product Name', 'Customer ID',
       'Customer Market', 'Customer Region', 'Customer Country',
       'Warehouse Country', 'Shipment Mode', 'Shipment Days - Scheduled',
       'Gross Sales', 'Discount %', 'Profit', 'Order Datetime',
       'Shipment Datetime', 'Order Processing Time', 'Shipment Delay'],
      dtype='object')

In [ ]:
df_orders[['Order Quantity', 'Shipment Days - Scheduled', 'Order Processing Time',
           'Discount %', 'Profit', 'Gross Sales', 'Shipment Delay']].describe()

Order Quantity  Shipment Days - Scheduled  Order Processing Time  \
count    30871.000000               30871.000000           30871.000000   
mean         2.149817                   3.072495               2.589777   
std          1.461393                   1.184417             131.195255   
min          1.000000                   1.000000            -976.000000   
25%          1.000000                   2.000000               1.000000   
50%          1.000000                   4.000000               2.000000   
75%          3.000000                   4.000000               4.000000   
max          5.000000                   4.000000             977.000000   

         Discount %        Profit   Gross Sales  Shipment Delay  
count  30871.000000  30871.000000  30871.000000    30871.000000  
mean       0.101296    129.383305    200.235690        0.482718  
std        0.070423     52.649857    114.251482      131.185163  
min        0.000000      6.000000     10.000000     -973.000000  
25%        0.040000     97.000000    120.000000       -1.000000  
50%        0.090000    125.000000    200.000000        1.000000  
75%        0.160000    150.000000    300.000000        3.000000  
max        0.250000    258.000000    533.000000      980.000000

## Order Processing Time

After checking the descriptive statistics on `df_orders`, we noticed some anomalies in the data. One of the issues identified is the presence of negative values in the "`Order Processing Time`" column. This finding suggests that certain products were shipped before they were even ordered, which does not align with the standard supply chain process.

In [ ]:
# Filtering the data to show only rows where the "Order Processing Time" is less than -1
filtered_data = df_orders[df_orders['Order Processing Time'] < -1]
filtered_data[['Order Item ID', 'Order Datetime', 'Shipment Days - Scheduled', 'Shipment Datetime',
               'Order Processing Time', 'Shipment Delay']].sort_values(by='Order Processing Time', ascending=False)

Order Item ID      Order Datetime  Shipment Days - Scheduled  \
4153          153501 2017-06-14 17:08:00                          4   
28770         172566 2017-10-12 13:49:00                          4   
28714         179160 2017-12-18 21:15:00                          4   
15541         114896 2016-11-17 15:30:00                          4   
28760         172950 2017-11-02 17:28:00                          2   
...              ...                 ...                        ...   
3271          169027 2017-09-13 21:02:00                          3   
5134          168862 2017-09-12 22:58:00                          2   
10388         169730 2017-09-17 23:50:00                          1   
19136         168167 2017-09-09 06:41:00                          4   
21805         168386 2017-09-10 09:39:00                          4   

      Shipment Datetime  Order Processing Time  Shipment Delay  
4153         2017-06-13                     -2               6  
28770        2017-10-11                     -2               6  
28714        2017-12-17                     -2               6  
15541        2016-11-16                     -2               6  
28760        2017-11-01                     -2               4  
...                 ...                    ...             ...  
3271         2015-02-19                   -938             941  
5134         2015-02-14                   -942             944  
10388        2015-02-09                   -952             953  
19136        2015-01-12                   -972             976  
21805        2015-01-09                   -976             980  

[2735 rows x 6 columns]

After further analysis and reviewing the original data, it was found that there was an unusual occurrence in the column "`Order Processing Time`" which was obtained by combining the columns "Order Year", "Order Month", etc., with a maximum value of `-977` days. This indicates that there are orders where the product was shipped approximately 2 years and 7 months before the order was placed, which is impossible in the real-world. There was no discernible pattern or explanation for this oddity, and that's how it was in the original data.

It was therefore decided to remove data points with negative "`Order Processing Time`". A total of 2735 data points out of 30871, about 8.85% of the data, were identified for deletion.

In [ ]:
# Drop the filtered data
df_orders.drop(df_orders[df_orders['Order Processing Time'] < -1].index, inplace=True)
df_orders.reset_index(drop=True, inplace=True)
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28136 entries, 0 to 28135
Data columns (total 21 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Order ID                   28136 non-null  int64         
 1   Order Item ID              28136 non-null  int64         
 2   Order YearMonth            28136 non-null  int64         
 3   Order Quantity             28136 non-null  int64         
 4   Product Department         28136 non-null  object        
 5   Product Category           28136 non-null  object        
 6   Product Name               28136 non-null  object        
 7   Customer ID                28136 non-null  int64         
 8   Customer Market            28136 non-null  object        
 9   Customer Region            28136 non-null  object        
 10  Customer Country           28136 non-null  object        
 11  Warehouse Country          28136 non-null  object        
 12  Ship

# Data Exporting & Conclusion

In [ ]:
# Final Check
df_orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28136 entries, 0 to 28135
Data columns (total 21 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Order ID                   28136 non-null  int64         
 1   Order Item ID              28136 non-null  int64         
 2   Order YearMonth            28136 non-null  int64         
 3   Order Quantity             28136 non-null  int64         
 4   Product Department         28136 non-null  object        
 5   Product Category           28136 non-null  object        
 6   Product Name               28136 non-null  object        
 7   Customer ID                28136 non-null  int64         
 8   Customer Market            28136 non-null  object        
 9   Customer Region            28136 non-null  object        
 10  Customer Country           28136 non-null  object        
 11  Warehouse Country          28136 non-null  object        
 12  Ship

In [ ]:
merged_inventory.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4200 entries, 0 to 4199
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Product Name             4200 non-null   object 
 1   Year Month               4200 non-null   int64  
 2   Warehouse Inventory      4200 non-null   int64  
 3   Inventory Cost Per Unit  4200 non-null   float64
 4   Average Demand           4025 non-null   float64
 5   Stock Coverage Ratio     4025 non-null   float64
 6   Total Inventory Cost     4200 non-null   float64
dtypes: float64(4), int64(2), object(1)
memory usage: 262.5+ KB


In [ ]:
merged_inventory['Product Name'].nunique()

118

In [ ]:
df_fulfillment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 118 entries, 0 to 117
Data columns (total 2 columns):
 #   Column                              Non-Null Count  Dtype  
---  ------                              --------------  -----  
 0   Product Name                        118 non-null    object 
 1   Warehouse Order Fulfillment (days)  118 non-null    float64
dtypes: float64(1), object(1)
memory usage: 2.0+ KB


## Exporting Cleaned Data

Save the cleaned and processed DataFrames (`df_orders`, `merged_inventory`) to an Excel file for further visualization and analysis in Power BI.

In [ ]:
# Export DataFrames to Excel
df_orders.to_excel('Orders.xlsx', index=False)
merged_inventory.to_excel('Inventory.xlsx', index=False)
df_fulfillment.to_excel('Fulfillment.xlsx', index=False)

## Conclusion

Through this analysis, we successfully explored and performed data cleaning on the `df_orders` and `df_inventory` DataFrames. Important metrics such as `Order Processing Time`, `Stock Coverage Ratio`, `Total Inventory Cost`, and `Shipment Delay` were calculated to understand inventory performance and supply chain efficiency for the company. Additionally, we removed some anomalous data points that could impact the analysis results. The cleaned and processed data will be exported to an Excel file for further analysis and visualization in Power BI.

The biggest challenge in this process was data validation and anomaly detection. Data may not always align with expectations and can contain errors or outliers. Ensuring the accuracy and validity of data while detecting anomalies is crucial for obtaining reliable analysis results.

To address these challenges, I tried to utilized various techniques and appropriate tools that I know for data validation and anomaly detection. Moreover, I conducted repeated checks between the original data to ensure that my analysis was based on accurate and reliable data.

# Closing

AAs an aspiring data analyst, I will consistently look for opportunities to improve my skills and insights. therefore, I respectfully ask and thank you for any suggestions or constructive criticism you may have regarding this analysis. Your input will certainly help me refine my analytical approach and gather a more comprehensive understanding of the data going forward.

This analysis was conducted as part of a competition organized by DataCamp, with the aim of demonstrating data analytics expertise and providing actionable insights to address real-world business challenges. Throughout this process, I strive to use effective data processing, data cleaning, and advanced analytics techniques to generate meaningful conclusions and support decision making.

I look forward to exploring more complex data analytics projects in the future. Once again, thank you for your time and support.

*   Email : farizalfitraaa@gmail.com
*   Linkedin : https://www.linkedin.com/in/farizalfitra/
*   Github : https://github.com/alfitraaa
